core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/EnvelopeCursor.java

```diff
@@ -367,12 +344,6 @@ public boolean findSpeed(double speed, CmpOperator operator) {
                 getPartBound(part, operator),
                 operator, speed);
 
-        // look for the next envelope part which contains a speed which matches the
-        // search requirement. The current envelope part may contain such a point,
-        // but it may be **before the cursor**.
-        if (!findPart(partPredicate))
-            return false;
-
```

Comment:
Is there a reason for this change? It sounds equivalent so I don't really care either way, but I may be missing something.

The comment in the following `while` becomes outdated with this change.

Reply:
There's likely a bug hidden somewhere, this block shouldn't do anything other than an optimization. It skips forward until it finds a part that can match the criteria.

Reply:
I can add it back, but it seemed very weird to me